### PR TITLE
x86 updates to create_llvm

### PR DESCRIPTION
**Pull request by alan-forbes-cp**
# Overview

Add x86 support to create_llvm

# Reason for change

x86 support

# Description of change

Add x86 llvm to cache. Tested in my fork.

### CODE REVIEW

File: .github/workflows/create_llvm.yml [hunk 1]
```diff
@@ -45,6 +45,8 @@ jobs:
           - arch_flags: -DCMAKE_TOOLCHAIN_FILE="$GITHUB_WORKSPACE/ock/platform/arm-linux/aarch64-toolchain.cmake"
                         -DLLVM_HOST_TRIPLE=aarch64-unknown-linux-gnu
             arch: aarch64
+          - arch_flags: -DLLVM_BUILD_32_BITS=ON -DLIBXML2_LIBRARIES=IGNORE -DLLVM_ENABLE_TERMINFO=OFF
```
Comment (hvdijk) [inline .github/workflows/create_llvm.yml:48]: This matches what we do on our other CI, but the second and third options are workaround for `LLVM_BUILD_32_BITS` being implemented in a way that doesn't handle everything that's needed, and these wouldn't be needed if we cross-compiled for 32-bit x86 in the same way that we do for AArch64. Maybe this is a good time to switch to that?

Does `LLVM_HOST_TRIPLE` need to be set too, or is that done automatically by `LLVM_BUILD_32_BITS`?

Comment (hvdijk) [reply .github/workflows/create_llvm.yml:52]: `LLVM_HOST_TRIPLE` is set for cross builds because the value that would be used implicitly is wrong, but that applies to 32-bit vs. 64-bit builds as well. I checked our internal CI now, in our current build with `-DLLVM_BUILD_32_BITS=ON`, the log includes:
```
-- LLVM host triple: x86_64-unknown-linux-gnu
...
-- LLVM default target triple: x86_64-unknown-linux-gnu
```
We probably get away with it because we only use this build in limited ways, and those limited ways do always set the target triple, but it's still wrong, this is not the right host or target triple for 32-bit x86.

Comment (coldav) [reply .github/workflows/create_llvm.yml:52]: if it's easy to do it in a cross way then sure, that makes sense - not sure how much Alan understands about it though.

Comment (alan-forbes-cp) [reply .github/workflows/create_llvm.yml:52]: I assume the triple required is `x86-unknown-linux-gnu` - in which case I can add in the same way as is done for aarch64.

Comment (hvdijk) [reply .github/workflows/create_llvm.yml:52]: It would be `i386-unknown-linux-gnu` or `i686-unknown-linux-gnu` depending on whether the build environment defaults to assuming newer processors. I do not know what the build environment in GitHub is like, @coldav do you know?

Comment (coldav) [reply .github/workflows/create_llvm.yml:52]: I don't know.

Comment (hvdijk) [reply .github/workflows/create_llvm.yml:52]: Okay, I checked, my local system's `gcc -v` includes `--with-arch-32=i686`, that's the Ubuntu-supplied GCC, and I think we use the Ubuntu-supplied GCC in GitHub too? It should be `i686-unknown-linux-gnu`, then.

File: .github/workflows/create_llvm.yml [hunk 2]
```diff
@@ -36,6 +36,8 @@ jobs:
             build_type: Release
           - os: windows-2019
             arch: aarch64
+          - os: windows-2019
```
Comment (coldav) [inline .github/workflows/create_llvm.yml:39]: you'll need to exclude more than that to get down to a single extra version

Comment (coldav) [reply .github/workflows/create_llvm.yml:39]: ah sorry I got confused

File: .github/workflows/create_llvm.yml (hunk 1 above)
Comment (alan-forbes-cp) [reply .github/workflows/create_llvm.yml:54]: Confirmed for Github - thanks @hvdijk

**Review by coldav — APPROVED** [review]

File: .github/workflows/create_llvm.yml [hunk 3]
```diff
@@ -88,10 +97,17 @@ jobs:
         run:
           sudo apt-get install --yes g++-11-aarch64-linux-gnu
 
+      - name: install x86 build tools
+        if: ${{ steps.cache.outputs.cache-hit != 'true' && matrix.arch == 'x86' }}
+        run: |
+          sudo dpkg --add-architecture i386
+          sudo apt-get update
+          sudo apt-get install --yes gcc-multilib g++-multilib libc6-dev:i386 lib32tinfo-dev
```
Comment (coldav) [inline .github/workflows/create_llvm.yml:105]: Your other change for setup_ubuntu installs this. So we may want to combine in some way at some point.

File: .github/workflows/create_llvm.yml [hunk 4]
```diff
@@ -18,20 +18,26 @@ jobs:
         version: [17, 18]
         os: [ubuntu-22.04, windows-2019]
         build_type: [Release, RelAssert]
-        arch : [x86_64, aarch64]
+        arch : [x86, x86_64, aarch64]
         exclude:
-          # Reduce the versions of llvm for aarch64 architecture, as
+          # Reduce the versions of llvm for aarch64 & x86 architectures, as
```
Comment (coldav) [inline .github/workflows/create_llvm.yml:23]: windows too, not part of this change as such, but since you are changing the line.

**Review by coldav — APPROVED** [review]

File: .github/workflows/create_llvm.yml (hunk 4 above)
Comment (alan-forbes-cp) [reply .github/workflows/create_llvm.yml]: Done

**Comment by alan-forbes-cp** [comment]
@codeplaysoftware/ock-workflow-reviewers
Could I have this one merged please? Ta.
